pkg/policy/rule_test.go

```diff
@@ -2575,3 +2576,40 @@ func (ds *PolicyTestSuite) TestMatches(c *C) {
 	c.Assert(hostRule.metadata.IdentitySelected, checker.DeepEquals,
 		map[identity.NumericIdentity]bool{selectedIdentity.ID: false, hostIdentity.ID: true})
 }
+
+func BenchmarkRuleString(b *testing.B) {
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		r := &rule{
+			Rule: api.Rule{
+				EndpointSelector: api.NewESFromLabels(labels.ParseSelectLabel("bar")),
+				Ingress: []api.IngressRule{
+					{
+						ToPorts: []api.PortRule{{
+							Ports: []api.PortProtocol{
+								{Port: "80", Protocol: api.ProtoTCP},
+								{Port: "8080", Protocol: api.ProtoTCP},
+							},
+							Rules: &api.L7Rules{
+								HTTP: []api.PortRuleHTTP{
+									{Method: "GET", Path: "/"},
+								},
+							},
+						}},
+					},
+				},
+				Egress: []api.EgressRule{
+					{
+						ToPorts: []api.PortRule{{
+							Ports: []api.PortProtocol{
+								{Port: "3000", Protocol: api.ProtoAny},
+							},
+						}},
+					},
+				},
+			},
+		}
```

Comment:
I'd suggest to move the rule construction out of the loop before `b.ReportAllocs`, so it doesn't influence benchmarks (namely allocations).

Reply:
Already update the code and benchmark result.